group_theory/free_group.lean

```diff
@@ -0,0 +1,360 @@
+-- https://math.stackexchange.com/a/852661/328173
```

Comment:
Add a header to the file similar to other files in mathlib, with you as the author